server/server.go

```diff
@@ -917,6 +911,9 @@ func (s *Server) DrainClients(drainWait time.Duration, cancelWait time.Duration)
 	go func() {
 		defer close(allDone)
 		for _, conn := range conns {
+			if !conn.getCtx().GetSessionVars().InTxn() {
```

Comment:
I am afrait that this is a compatibility breaker. /cc @djshow832

Reply:
The behaviour is same as https://github.com/pingcap/tidb/pull/32111 merged before, like v6.5.x

Reply:
Why do you think it's a compatibility breaker? @xhebox

Reply:
It is different from 7.1, but same as 6.5.